extensions/blocks/dialogue/edit.js

```diff
@@ -259,16 +284,38 @@ export default function DialogueEdit ( {
 						return createBlock( blockNameFallback );
 					}
 
-					if ( ! value ) {
-						return createBlock( blockName );
-					}
-
 					return createBlock( blockName, {
 						...attributes,
 						content: value,
 					} );
 				} }
-				onReplace={ onReplace }
+				onReplace={ ( blocks, ...args ) => {
```

Comment:
Recommend a small comment at the top of this explaining its purpose; if I came back in 6 months and looked at it I might be confused.